lib/ex_machina/ecto.ex

```diff
@@ -141,4 +190,10 @@ defmodule ExMachina.Ecto do
     |> Enum.reject(fn({_, value}) -> value == nil end)
     |> Enum.into(%{})
   end
+
+  defp convert_atom_keys_to_strings(struct) do
+    for {key, value} <- struct,
```

Comment:
@pdawczak I think this needs to be expanded to work for nested maps. Can you also add a test case for that?

Also, minor, but would you mind switching this to use `Enum.reduce` (or something in that family).